src/main/java/recipeapplication/controllers/RecipeController.java

```diff
@@ -38,4 +34,21 @@ private Recipe toEntity(RecipeTO recipe) {
 
 		return recipeEntity;
 	}
+
+	@RequestMapping(value = "/putRecipeById", method = RequestMethod.PUT)
+	public void recipePut(@RequestParam String id,
+						   @RequestPart(value = "recipeImage", required = false) MultipartFile file,
+						  @ModelAttribute("recipe") RecipeTO recipe) {
+		if (id.equalsIgnoreCase(recipe.getId()) &&
+				recipe.getCategory() != null &&
+				recipe.getIngridiant() != null &&
```

Comment:
instead of checking for Null properties, you can use the annotation @Valid next to RecipeTO recipe and it would validat the properties (since each property have the @NotNull annotation)

Reply:
Has been edited to:

public void recipePut(@RequestParam String id,
						   @RequestPart(value = "recipeImage", required = false) MultipartFile file,
						  @ModelAttribute("recipe") @Valid RecipeTO recipe ) {
		if (id.equalsIgnoreCase(recipe.getId()))
			this.recipes.updateRecipe(toEntity(recipe));
		}

Reply:
Use @PathVariable("id") instead of @RequestParam for the id (the route should be "/recipe/{id}"